File: apps/backend/internal/model/conversation.go
package model

import (
	"time"
)

type Conversation struct {
	ID            uint       `gorm:"primarykey" json:"id"`
	Type          string     `gorm:"type:enum('single','group');default:'single';index" json:"type"`
	User1ID       *uint      `gorm:"index" json:"user1_id,omitempty"`
	User2ID       *uint      `gorm:"index" json:"user2_id,omitempty"`
	GroupID       *uint      `gorm:"index" json:"group_id,omitempty"`
	LastMessageID *uint      `json:"last_message_id,omitempty"`
	LastMessageAt *time.Time `gorm:"index" json:"last_message_at,omitempty"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
	
	// 关联
	User1       *User    `gorm:"foreignKey:User1ID" json:"user1,omitempty"`
	User2       *User    `gorm:"foreignKey:User2ID" json:"user2,omitempty"`
	Group       *Group   `gorm:"foreignKey:GroupID" json:"group,omitempty"`
	LastMessage *Message `gorm:"foreignKey:LastMessageID" json:"last_message,omitempty"`
}

func (Conversation) TableName() string {
	return "conversations"
}

// ConversationType 常量
const (
	ConversationTypeSingle = "single"
	ConversationTypeGroup  = "group"
)

File: apps/android/app/src/main/java/com/lanxin/im/data/model/Conversation.kt
package com.lanxin.im.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "conversations")
data class Conversation(
    @PrimaryKey
    val id: Long,
    val type: String, // single, group
    val user1Id: Long?,
    val user2Id: Long?,
    val groupId: Long?,
    val lastMessageId: Long?,
    val lastMessageAt: Long?,
    val unreadCount: Int = 0,
    val createdAt: Long,
    val updatedAt: Long,
    // UI展示字段
    val avatar: String? = null,
    val name: String? = null,
    val lastMessage: String? = null,
    val lastMessageTime: Long = updatedAt,
    val lastMessageType: String = "text",
    val lastMessageContent: String = lastMessage ?: "",
    val isMuted: Boolean = false,
    val isTop: Boolean = false,
    val draft: String? = null
)

object ConversationType {
    const val SINGLE = "single"
    const val GROUP = "group"
}

